azure-configurationservice/azure/configuration/aio/azure_configuration_client_async.py

```diff
@@ -0,0 +1,262 @@
+            pipeline=self._create_azconfig_pipeline(),
+        )
+
+    def _create_azconfig_pipeline(self):
+        policies = [
+            self.config.headers_policy,
+            self.config.user_agent_policy,
+            self.config.logging_policy,  # HTTP request/response log
+            AzConfigRequestsCredentialsPolicy(self.config.credentials),
+        ]
+
+        return AsyncPipeline(
+            AsyncioRequestsTransport(self.config),  # Send HTTP request using requests
+            policies,
+        )
+
+    def list_configuration_settings(
+        self, labels=None, keys=None, accept_date_time=None, fields=None, **kwargs
+    ):  # type: (list, list, datetime, list, dict) -> Paged
+        """
+        Example
+
+        .. code-block:: python
+
+            from datetime import datetime, timedelta
+
+            accept_date_time = datetime.today() + timedelta(days=-1)
+
+            all_listed = async_client.list_configuration_settings()
+            async for item in all_listed:
+                pass  # do something
+
+            filtered_listed = async_client.list_configuration_settings(
+                labels=["*Labe*"], keys=["*Ke*"], accept_date_time=accept_date_time
+            )
+            async for item in filtered_listed:
+                pass  # do something
+        """
+        return super().list_configuration_settings(
+            labels, keys, accept_date_time, fields, **kwargs
+        )
+
+    async def get_configuration_setting(
+        self, key, label=None, accept_date_time=None, **kwargs
+    ):
+        """
+        Example
+
+        .. code-block:: python
+
+            # in async function
+            fetched_config_setting = await async_client.get_configuration_setting(
+                key="MyKey", label="MyLabel"
+            )
+        """
+        return await super().get_configuration_setting(
+            key, label=label, accept_date_time=accept_date_time, **kwargs
+        )
+
+    async def add_configuration_setting(self, configuration_setting, **kwargs):
+        # type: (ConfigurationSetting, dict) -> ConfigurationSetting
+        """
+        Example
+
+        .. code-block:: python
+
+            # in async fuction
+            config_setting = ConfigurationSetting(
+                key="MyKey",
+                label="MyLabel",
+                value="my value",
+                content_type="my content type",
+                tags={"my tag": "my tag value"}
+            )
+            added_config_setting = await async_client.add_configuration_setting(config_setting)
+        """
+        return await super().add_configuration_setting(configuration_setting, **kwargs)
+
+    async def update_configuration_setting(
+        self,
+        key,
+        value=None,
+        content_type=None,
+        tags=None,
+        label=None,
+        etag=None,
+        **kwargs
+    ):
+        # type: (str, str, str, dict, str, str, dict) -> ConfigurationSetting
```

Comment:
Isn't the tags dict in this case a Mapping[str, str]?

Reply:
yes. Do you mean the type hint should be dict[str, str]? Will update it